src/main/java/homework_6/Main.java

```diff
@@ -0,0 +1,30 @@
+package homework_6;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Main {
+
+    public static void main(String[] args) {
+        mapCollisionProblems();
+        mapMutableProblems();
+    }
+
+    public static void mapCollisionProblems() {
+        MapProblemsCollisionGenerator mpcg1 = new MapProblemsCollisionGenerator("Generator 1");
+        MapProblemsCollisionGenerator mpcg2 = new MapProblemsCollisionGenerator("Generator 2");
+        Map<MapProblemsCollisionGenerator, Integer> collisionMap = new HashMap<>();
+        collisionMap.put(mpcg1, 1);
+        collisionMap.put(mpcg2, 2);
+        System.out.println(collisionMap.toString());
```

Comment:
Opt: how do you know there's collision? You could as example add sout to equals, so you'll know that collision appeared